lib/datadog/appsec/contrib/sinatra/patcher.rb

```diff
@@ -55,7 +57,9 @@ def dispatch!
 
             # TODO: handle exceptions, except for super
 
-            request_return, request_response = Instrumentation.gateway.push('sinatra.request.dispatch', request) do
+            gateway_request = Datadog::AppSec::Contrib::Rack::Gateway::Request.new(env)
```

Comment:
WDYT of having a `Contrib::Sinatra::Gateway::Request` inheriting from `Contrib::Rack::Gateway::Request`? instead of depending on the latter directly?